docs/HazardAnalysis/HazardAnalysis.tex

```diff
@@ -72,17 +76,121 @@ \section{Critical Assumptions}
-				\hline
-			\end{tabular}
-		\end{adjustbox}
-	\end{table}
+	\begin{longtable}{|p{0.15\textwidth}|p{0.15\textwidth}|p{0.25\textwidth}|p{0.25\textwidth}|p{0.35\textwidth}|p{0.1\textwidth}|p{0.05\textwidth}|}
+		\caption{Failure Mode and Effect Analysis Table}                                                                \\
+		\hline
+		\multicolumn{1}{|c|}{\textbf{Component}}
+		 & \multicolumn{1}{|c|}{\textbf{Failure Modes}}
+		 & \multicolumn{1}{|c|}{\textbf{Effects of Failure}}
+		 & \multicolumn{1}{|c|}{\textbf{Causes of Failure}}
+		 & \multicolumn{1}{|c|}{\textbf{Recommended Action}}
+		 & \multicolumn{1}{|c|}{\textbf{SR}}
+		 & \multicolumn{1}{|c|}{\textbf{Ref.}}                                                                          \\
+		\hline
+		\multirow{2}{*}{Work Orders}
+		 & Work Order is missing
+		 & Customer and Employees will not know any of the work that has been done for a specific job
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Database failure
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Regular and automatic database backups/snapshots and allow shop owners to request rollbacks
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item
+		   \end{enumerate}
+		 & H5-1                                                                                                         \\
+		\cline{2-7}
+		~
+		 & Work Order is missing detailed information
+		 & Customer and Employees will not know all of the work that has been done for a specific job
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Database failure
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Refer to H5-1a
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item
+		   \end{enumerate}
+		 & H5-2                                                                                                         \\
+		\hline
+		Shop Profile
+		 & Unable to find details of a shop
+		 & Customers will not be able to see the information about a specific shop (e.g., address, phone number, etc.)
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Database failure
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Refer to H5-1a
+		   \end{enumerate}
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item
+		   \end{enumerate}
+		 & H6-1                                                                                                         \\
+		\hline
+		\multirow{2}{*}{Services}
+		 & Unable to find a service
+		 & Customers and Employees will not be able to see what services are offered by the shop
+		 & \begin{enumerate}[label=\alph*., leftmargin=*]
+			   \item Database failure
```

Comment:
could this be due to another failure? e.g., a bug?

Reply:
I couldn't think of any others. Our app's functionality is really simple, we just have a large feature set. I don't think "bug" is a good failure cause because it is really vague. Also not sure what recommended action and requirement we would put for that besides "The app shall be tested". I don't think saying that we will have tests is a good requirement since it does not guarantee that there isn't a problem and wouldn't provide closure to this hazard.

Reply:
Network failure? I assume that if their network is down, the calls to external services or DB won't work?

Reply:
So we can say:
Failure cause: Client-side network failure
Recommended Action: Notify User they are offline and may be missing search results

and the requirement @TimChoy can have is "notify user when they are offline".

Reply:
Or do we want to disable searching when offline?

Reply:
We didn't specifically say anything about offline requirements, though you did talk about the fact that this is a PWA and should have offline capabilities.

Reply:
Yea that is what I was thinking. We didn't add offline capabilities into the scope of this project. I left the recommended action to just notify that they are offline. I don't want to add a lot more scope to this.

Reply:
Notifying that they are offline should be enough for this project.